tools/universe/package_manager.py

```diff
@@ -9,129 +9,64 @@
 import json
 import os
 import tempfile
+import urllib.request
 
 from . import package
-try:
-    import requests
-    _HAS_REQUESTS = True
-except ImportError:
-    _HAS_REQUESTS = False
 
 
 LOGGER = logging.getLogger(__name__)
 
 
 class PackageManager:
     """A simple package manager for retrieving universe packages"""
-    def __init__(self, universe_url="https://universe.mesosphere.com/repo",
-                 dcos_version="1.10",
+    def __init__(self,
+                 universe_package_prefix="https://universe.mesosphere.com/package/",
+                 dcos_version="1.11",
                  package_version="4",
                  dry_run=False):
 
         self._dry_run = dry_run
-        self._universe_url = universe_url
+        self._universe_package_prefix = universe_package_prefix
         self._headers = {
             "User-Agent": "dcos/{}".format(dcos_version),
             "Accept": "application/vnd.dcos.universe.repo+json;"
                       "charset=utf-8;version=v{}".format(package_version),
         }
 
-        self.__package_cache = None
+        self.__package_cache = {}
 
-        if _HAS_REQUESTS:
-            self._get_packages = _get_packages_with_requests
-        else:
-            self._get_packages = _get_packages_with_curl
 
     def get_package_versions(self, package_name):
         """Get all versions for a specified package"""
+        if self._dry_run:
+            return DryRunPackages()
+
+        if package_name not in self.__package_cache:
+            LOGGER.info("Retrieving information for package: %s", package_name)
+            url = self._universe_package_prefix + package_name
```

Comment:
We could use `urllib.parse.urljoin(self._universe_package_prefix, package_name)` instead?

Reply:
Good point, done.